tfhe/src/c_api/high_level_api/keys.rs

```diff
@@ -6,18 +6,34 @@ pub struct PublicKey(pub(crate) crate::high_level_api::PublicKey);
 pub struct CompactPublicKey(pub(crate) crate::high_level_api::CompactPublicKey);
 pub struct CompressedCompactPublicKey(pub(crate) crate::high_level_api::CompressedCompactPublicKey);
 pub struct ServerKey(pub(crate) crate::high_level_api::ServerKey);
+/// Compressed version of the ServerKey
+///
+/// Allows to save storage space and transfer time.
+/// Also, the CompressedServerKey is the key format that allows to select
+/// the target hardware of the actual ServerKey when decompressing it.
+pub struct CompressedServerKey(pub(crate) crate::high_level_api::CompressedServerKey);
+
+/// ServerKey that lives on a Cuda GPU
+#[cfg(feature = "gpu")]
+pub struct CudaServerKey(pub(crate) crate::high_level_api::CudaServerKey);
 
 impl_destroy_on_type!(ClientKey);
 impl_destroy_on_type!(PublicKey);
 impl_destroy_on_type!(CompactPublicKey);
 impl_destroy_on_type!(CompressedCompactPublicKey);
 impl_destroy_on_type!(ServerKey);
+impl_destroy_on_type!(CompressedServerKey);
+impl_destroy_on_type!(
+    #[cfg(feature = "gpu")]
```

Comment:
I guess you could have this guard on the macro directly

Reply:
Yes, I think I made it like it because I was having some errors about the CudaCompressedServerKey, but they were caused by something else, will fix